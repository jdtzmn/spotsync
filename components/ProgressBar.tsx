import React from 'react';

interface ProgressProps {
  percent: number,
  color?: string,
  height?: number
}

const ProgressDefaults = {
  color: '#403EBE',
  height: 2
}

const Progress = ({ percent, color, height, ...otherProps }: ProgressProps) => {
  const defaultStyles = {
    backgroundColor: color || ProgressDefaults.color,
    height: height || ProgressDefaults.height,
    width: percent ? `${percent}%` : 0
  }

  return (
    <div {...otherProps} style={defaultStyles} />
  )
}

export default Progress
