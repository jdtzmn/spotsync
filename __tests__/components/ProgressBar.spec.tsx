import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import ProgressBar from 'components/ProgressBar'

describe('Progress', () => {
  let wrapper: ShallowWrapper
  beforeEach(() => {
    wrapper = shallow(<ProgressBar percent={0} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the correct default props', () => {
    expect(wrapper.prop('style'))
      .toEqual({
        backgroundColor: '#403EBE',
        height: 2,
        width: 0
      })
  })

  it('should render according to props', () => {
    wrapper.setProps({
      color: 'blue',
      height: 4,
      percent: 5
    })

    expect(wrapper.prop('style'))
      .toEqual({
        backgroundColor: 'blue',
        height: 4,
        width: '5%'
      })
  })

  it('should allow custom props to be added to the div', () => {
    wrapper.setProps({ test: 'prop' })

    expect(wrapper.prop('test')).toBe('prop')
  })
})
