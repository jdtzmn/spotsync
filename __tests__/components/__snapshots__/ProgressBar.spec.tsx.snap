// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should render correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#403EBE",
      "height": 2,
      "width": 0,
    }
  }
/>
`;
